compiler/rustc_mir_build/src/build/expr/as_place.rs

```diff
@@ -416,66 +406,63 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         let this = self;
         let expr_span = expr.span;
         let source_info = this.source_info(expr_span);
-        match expr.kind {
+        match &expr.kind {
             ExprKind::Scope { region_scope, lint_level, value } => {
-                this.in_scope((region_scope, source_info), lint_level, |this| {
-                    let value = this.hir.mirror(value);
-                    this.expr_as_place(block, value, mutability, fake_borrow_temps)
+                this.in_scope((*region_scope, source_info), *lint_level, |this| {
+                    this.expr_as_place(block, &value, mutability, fake_borrow_temps)
                 })
             }
             ExprKind::Field { lhs, name } => {
-                let lhs = this.hir.mirror(lhs);
-                let place_builder =
-                    unpack!(block = this.expr_as_place(block, lhs, mutability, fake_borrow_temps,));
-                block.and(place_builder.field(name, expr.ty))
+                let place_builder = unpack!(
+                    block = this.expr_as_place(block, &lhs, mutability, fake_borrow_temps,)
+                );
+                block.and(place_builder.field(*name, expr.ty))
             }
             ExprKind::Deref { arg } => {
-                let arg = this.hir.mirror(arg);
-                let place_builder =
-                    unpack!(block = this.expr_as_place(block, arg, mutability, fake_borrow_temps,));
+                let place_builder = unpack!(
+                    block = this.expr_as_place(block, &arg, mutability, fake_borrow_temps,)
+                );
                 block.and(place_builder.deref())
             }
             ExprKind::Index { lhs, index } => this.lower_index_expression(
                 block,
-                lhs,
-                index,
+                &lhs,
+                &index,
```

Comment:
due to the reference taken at the match level, this reference should be unnecessary